### PR TITLE
add basic store

### PR DESCRIPTION
**Pull request by andriiivaniuk**

### CODE REVIEW

**Comment by andriiivaniuk** [comment]
> Does that directory path works correctly?

it does :)

**Review by nastiaymlnv — DISMISSED** [review]

File: src/ducks/store.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { legacy_createStore, applyMiddleware } from "redux";
+import localStorageMiddleware from "../middleware/localStorageMiddleware";
+import { composeWithDevTools } from "redux-devtools-extension";
+import reducers from ".";
```
Comment (vladhybalo) [inline src/ducks/store.jsx]: Interested how does it work)

File: src/ducks/store.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { legacy_createStore, applyMiddleware } from "redux";
+import localStorageMiddleware from "../middleware/localStorageMiddleware";
+import { composeWithDevTools } from "redux-devtools-extension";
```
Comment (vladhybalo) [inline src/ducks/store.jsx]: pls align import in some logical order
I suggest:
1st group - libraries import
2nd group - store and ducks
3rd group - components and hocs
4th group - local files

File: src/ducks/testState/testStateReducer.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { ADD_TEST_VALUE, INIT_TEST_STATE } from "./testStateActions";
+
+export const testStateReducer = (state = {}, action) => {
+    
+    switch (action.type) {
+        case INIT_TEST_STATE:
+            console.log("INIT_TEST_STATE");
```
Comment (vladhybalo) [inline src/ducks/testState/testStateReducer.jsx]: Let's have a general rule to remove all unused code and console logs

File: src/middleware/localStorageMiddleware.jsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+
+export const localStorageMiddleware = (store) => (next) => (action) =>  {
+
+    next(action);
+
```
Comment (vladhybalo) [inline src/middleware/localStorageMiddleware.jsx]: redundant enters

File: src/middleware/localStorageMiddleware.jsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
```
Comment (vladhybalo) [inline src/middleware/localStorageMiddleware.jsx]: I think it make sense to rename folder middleware to middleware**s**

File: src/ducks/store.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+const getInitialStore = () => {
+    return {
+        testState: {}
+    }
+}
+
+export const store = legacy_createStore(
+    reducers,
+    getInitialStore(),
+    composeWithDevTools(applyMiddleware(localStorageMiddleware))
```
Comment (vladhybalo) [inline src/ducks/store.jsx]: I will add it to story description, lets also support thunk middleware in this pr

File: src/ducks/store.jsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { legacy_createStore, applyMiddleware } from "redux";
```
Comment (vladhybalo) [inline src/ducks/store.jsx]: this file should be in the `src` folder as it doesn't rely to ducks. Ducks are about actions, action creators and reducers

**Review by vladhybalo — DISMISSED** [review]

**Review by nastiaymlnv — DISMISSED** [review]

**Review by nastiaymlnv — DISMISSED** [review]

**Review by nastiaymlnv — APPROVED** [review]
Good job!

**Review by vladhybalo — APPROVED** [review]

**Pull request merged**